### ISSUE TITLE
test(model): verify one V4.1 compressed source-reader pair numerically

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13323]
# test(model): verify one V4.1 compressed source-reader pair numerically

```routing
repo: fak
lane: model
paths: ["internal/model/v41_source_reader_oracle_test.go","internal/model/v41_source_reader_fixture_test.go"]
expected_steps: 3
```

## Working spine
Expose V4.1 integration failures before another full-model hardware run, using existing runtime and test surfaces.

## Current state
v41ParityStagesModel deliberately places the compressed/index source in the final layer so no later reader requires publication (v41_parity_test.go:306-373). TestV41ParityStages verifies execution, finite output and sensitivity to disabling stages, not independent numerical parity for a source followed by a reader. Existing component/state tests remain useful and must be reused.

## Why this is next
The native mission has reached a token but continues through expensive sequential bring-up. This leaf removes one class of uncertainty in a bounded software check while the current performance work proceeds.

## Parent context
anthony-chaudhary/fak#12640; engineering support for anthony-chaudhary/fak#13294 and MISSION-halo-v41-native. Existing deepseek-v41-flash and public issue missions can dispatch this single-package leaf.

## Core through-line
Add one tiny two-layer ratio-2 source/reader fixture through the existing session/forward path. Independently compute the compressed rows, selected row IDs and reader attention result using pinned reference equations, then compare production outputs for token counts immediately below, at and above one compression boundary. Isolate this source-reader interaction; deterministic small tensors avoid hardware and full checkpoint loading.

## Gold-plating boundary
Ratio 2, one source and one reader only; no full 40-layer reference, ratio-1/128 expansion, new retained-decode API, Engram, incremental decode implementation or production change. This is an independent correctness witness for existing full-prefix semantics; #13305-13313 retain the incremental state implementation.

## Worker packet
DS4.1 or Qwen3.8 capability level; S1. Own only the paths below. Three steps: establish the specified failing/missing contract with an independent fixture; implement only the bounded change; run the named and owning-package gates and land with explicit paths. No open-ended study or architecture choice remains. If a source symbol moved, locate its equivalent; do not silently widen the contract.

## Work unit
leaf

## Assumptions
- The pinned V4.1 architecture and existing production/test seams remain the authority; re-read changed symbols before editing.

## Confusion risks
- A passing synthetic witness proves the declared software envelope only, not physical performance or complete checkpoint equivalence.

## Coordination
- Acquire the listed public lane and preserve peer work. Serialize shared model-lane edits; use the dependencies section for start holds.

## Trigger
Operator-requested architecture planning after repeated sequential native bring-up failures; source-backed gaps documented in Current state.

## Batch policy
One issue per unique fak-native-shiftleft-key; search open and closed tickets before filing or expanding. Update the existing leaf on rerun.
## Work scale
S1

## Model tier
Required: T1
Optimal: T1

## Dependencies and scope fences
- Coordinates with: anthony-chaudhary/fak#13305 and anthony-chaudhary/fak#13309 (state optimizations, not a prerequisite).
- Start blocked by: none.

## Reference and reuse decision
ADAPT the [pinned V4.1 Attention/compression reference](https://huggingface.co/deepseek-ai/DeepSeek-V4.1-Flash/blob/dba1be0a40aa45a94ad051997016db3960a90277/inference/model.py) (MIT) and reuse existing independent scalar primitives. Authority: Compressor.forward, Indexer.forward and Attention.forward in the pinned model.py. Fixture metadata records source SHA, exact ModelArgs/config digest, seed/dtype, input and weight digests, schedule and position convention. Expected rows, IDs and reader outputs must not call v41CompressedRows, v41IndexRows, production RMS/matmul helpers or consume production-generated intermediates. Use independently transcribed scalar math.

## Reproduction witness
Source-backed coverage gap: inspect the functions and current tests named above at public revision 3ce59641b971a31b6bf19838290e64c0bf5d1033, then add the named regression. A missing test is not a passing baseline. Capture its RED result or explicit missing-coverage finding before changes; do not claim an unexecuted production failure.

## Done condition / witness
Witness: `go test ./internal/model -run '^TestV41SourceReaderNumericOracle$' -count=1 -v`
The test is a required deliverable, not asserted to exist today. Confirm the exact test appears in `go test ./internal/model -list '^TestV41SourceReaderNumericOracle$'`; empty matches and unexpected skips fail acceptance. Then run `go test ./internal/model -count=1` and required boundary gates after the last edit, in the provisioned public candidate worktree.

## Verifiable Witness
```
go test ./internal/model -run '^TestV41SourceReaderNumericOracle$' -count=1 -v
```

## Definition of done
- [ ] Lengths 1, 2 and 3 independently assert compressed-row count, selected IDs and numeric reader output with explicit tolerance; source and reader use distinguishable inputs.
- [ ] A negative control with a wrong source layer or stale published row is detected; merely changing logits or observing finite values is insufficient.
- [ ] Actual production source publication and reader consumption execute, and focused plus owning-package tests pass without a checkpoint download.

## Acceptance gate
Every checklist item is required; focused tests plus owning-package tests and independent review must pass. This is [SW-VERIFIED] coverage only. Physical throughput and coherent two-prompt qualification remain open with the existing mission owner.

## Blast radius and fallback
One package: internal/model. Existing behavior remains the fallback until this leaf passes. Preserve unrelated work, fail closed on unsupported shapes, and retain the prior physical mission gate. No new hardware run is a prerequisite.

## Closure binding
DCO sign-off, explicit paths, issue reference, and (fak model) commit subject suffix. Follow existing provenance and lease rules. Land via Fak Flow and push; no agent-created PR. Closure covers this leaf only.

## Likely files
- internal/model/v41_source_reader_oracle_test.go
- internal/model/v41_source_reader_fixture_test.go

## Expected steps
3

## Value
- Centrality: Core
- P1 Context: advanced - turns a known integration gap into a reusable local check
- P2 Net value: advanced - avoids repeated full-model diagnosis without relaxing correctness
- P3 Adaptation: preserved - reuses existing contracts and keeps runtime policy stable
- P4 Operations: advanced - bounded inputs and exact failure evidence support ordinary workers

## Work estimate
Estimate: 3 points

## Overall completion contribution
Contribution: 3/30 points

## Completion standard
development

## Qualification target
Pinned V4.1 text architecture; scoped synthetic or metadata-only software gate on a CPU development host.

## Planning baseline
Source inspection and planning only at filing; implementation and its tests are not yet witnessed.

**Comment by anthony-chaudhary** [comment]
﻿Landed as `e10dff2bc4bf5f997d9ca593e72f14b730430dbc` on `origin/main` (commit subject: `test(model): verify one V4.1 compressed source-reader pair numerically (fak model)`).

## What landed
Two test-only files in `internal/model`:
- `v41_source_reader_fixture_test.go` — deterministic ratio-2 source/reader fixture plus an independently transcribed scalar reference (`v41SRRefProject` / `v41SRRefPool` / `v41SRRefScores` / `v41SRRefSelectRows`) that calls no production helper.
- `v41_source_reader_oracle_test.go` — `TestV41SourceReaderNumericOracle`, driving the real production `v41CompressedRows` -> `NewV41IndexerPublication` -> `Reuse` seam.

## Witness (green at landed tip)
```
go test ./internal/model -run '^TestV41SourceReaderNumericOracle$' -count=1 -v
=> PASS (6 subtests, 0 skips)
go test ./internal/model -count=1 -skip '^TestVerbalizableDirectionLiveArtifact$' => ok (257s)
go vet ./internal/model => clean
fak-boundary check --staged (5 gates) => PASS, 0 violations
```
The excluded test (`TestVerbalizableDirectionLiveArtifact`) fails only in this isolated worktree because its `experiments/verbalizable-direction-qwen25-0.5b/*` artifacts are skip-worktree (sparse). It passes on the root checkout; unrelated to this change.

## Scope notes
Lengths 1/2/3 span one compression boundary (ratio 2) and assert compressed-row count, selected IDs and the reader's numeric output against the independent reference, with two non-vacuous negative controls: a stale publication whose stored candidate mask must constrain the reader, and a wrong-source-layer resolution via the real `v41AttentionSourceUpdates` + `KVSourceRows` seam.

The reference selection was aligned to the production contract: `V41SelectIndexRows` emits exactly `min(topK, len(logits))` entries and pads (via `-1`) only for masked/invisible slots that survive the top-k; padding a short selection to the consumer's fixed stride is `v41AttentionIndexList.one`'s job, not the selector's. An earlier in-progress attempt asserted topK-length padding at the selector and was RED; this landed form matches the seam and is green.

[SW-VERIFIED] only; no physical throughput or checkpoint-equivalence claim.

**Comment by anthony-chaudhary** [comment]
Resolved by e10dff2bc4bf5f997d9ca593e72f14b730430dbc (landed on origin/main); witness green as recorded above.
